### PR TITLE
Component development section update

### LINKED ISSUE #518: [NEW] - Creating custom components
## Describe the issue briefly.

The documentation on component creation is from internal export. It needs to be evaluated and rewritten to be up to date and match our style.

## Which page(s) / section(s) are affected?

Requires update:

* [Custom Components](https://o3deorg.netlify.app/docs/user-guide/components/development/create-component/)
* [Reflecting Components](https://o3deorg.netlify.app/docs/user-guide/components/development/reflection/)

These pages can likely be retired and rolled into information on creating custom components in general, and replaced with samples or tutorials:

* [Editor Components](https://o3deorg.netlify.app/docs/user-guide/components/development/entity-system-pg-editor-components/) 
* [System Components](https://o3deorg.netlify.app/docs/user-guide/components/development/system-wide/)

## Does this work have an engineering dependency? What is it?

Unknown and unlikely.

### PR DESCRIPTION
**Pull request by willihay**
Signed-off-by: William Hayward <wilhayw@amazon.com>

## Change summary

Includes many improvements to the Programming - Component Development section, including:
* Rewrite of the component overview.
* Rewrite of the procedural/tutorial on creating a component.
* Modernization, clarifications, and corrections made to most topics.
* Style and formatting fixes made to older information.

Note:
* There's still some work to do in the topic on System Components. I'm still learning about that area. Updates to that topic will go in this PR if there's time, otherwise I will create a new issue for a future PR.

### Submission Checklist:

* [x] **Descriptive active voice** - Do descriptive sentences have a clear *subject* and *action verb*?
* [x] **Answer the question at hand** - Does the documentation answer a *what*, *why*, *how*, or *where* type of question?
* [x] **Consistency** - Does the content consistently follow the [style guide](https://o3de.org/docs/contributing/to-docs/style-guide/)?
* [x] **Help the user** - Does the documentation show the user something *meaningful*?

Closes #518 

@netlify /docs/user-guide/programming/components

### CODE REVIEW

**Review by micronAMZN — APPROVED** [review]
Approving pending technical review with a couple very minor comments.

**Comment by micronAMZN** [comment]
Great work.

File: content/docs/user-guide/programming/components/create-component.md
Comment (willihay) [reply content/docs/user-guide/programming/components/create-component.md:19]: Agreed. I will remove this boilerplate code and replace it with instructions on how to use the new template. However, that doc update will be done in the development branch, after this PR is merged to main and brought over to development.

**Comment by willihay** [comment]
Setting a do-not-merge label because this PR now has outdated information. It should be updated to reflect the new default component template - removing the default component boilerplate code and adding the o3de script command to create a component from template.

File: content/docs/user-guide/programming/components/create-component.md
Comment (willihay) [inline content/docs/user-guide/programming/components/create-component.md:12]:
```suggestion
1. Create the component class from the default component template, or from an existing component.
```

File: content/docs/user-guide/programming/components/create-component.md
Comment (willihay) [reply content/docs/user-guide/programming/components/create-component.md:19]: Update: since this PR is still open, I will update these instructions on creating a component using the new template.

File: content/docs/user-guide/programming/components/create-component.md
Comment (willihay) [inline content/docs/user-guide/programming/components/create-component.md:10]:
```suggestion
While every component is unique, the typical steps for creating a new component include the following:
```

File: content/docs/user-guide/programming/components/create-component.md
Comment (JwGedit) [inline content/docs/user-guide/programming/components/create-component.md:160]:
```suggestion
        (Required) Called when the owning entity is activated, provided that all services and components that the component depends on are present and active. The `Activate` function is always called _after_ any components that it depends on. To learn how to specify dependencies, refer to [Defining and Using Component Services](services). Typically in the `Activate()` function, a component performs setup procedures, connects to Event Buses (EBuses), and allocates resources or requests assets.
```

**Review by JwGedit — CHANGES_REQUESTED** [review]
Edit pass complete.

File: content/docs/user-guide/programming/components/create-component.md
Comment (willihay) [inline content/docs/user-guide/programming/components/create-component.md:131]:
```suggestion
1. Use the `AZ_COMPONENT` macro to define a universally unique identifier (UUID) for your component. The macro takes two arguments:
```

File: content/docs/user-guide/programming/components/editor-components.md
Comment (willihay) [reply content/docs/user-guide/programming/components/editor-components.md:15]: Added italics to "edit time". We've started italicizing terms right before we define them.

File: content/docs/user-guide/programming/components/create-component.md
Comment (willihay) [inline content/docs/user-guide/programming/components/create-component.md:160]:
```suggestion
        (Required) Called when the owning entity is activated, provided that all services and components that the component depends on are present and active. The `Activate` function is always called _after_ any components that it depends on. For information about how to specify dependencies, refer to [Defining and Using Component Services](services). Typically in the `Activate()` function, a component performs setup procedures, connects to Event Buses (EBuses), and allocates resources or requests assets.
```

File: content/docs/user-guide/programming/components/create-component.md
Comment (willihay) [reply content/docs/user-guide/programming/components/create-component.md]: Had to apply this separately since it was a second suggestion for the same line.

File: content/docs/user-guide/programming/components/create-component.md
Comment (willihay) [reply content/docs/user-guide/programming/components/create-component.md:19]: @lumberyard-employee-dm FYI, this PR now has instructions for creating the component from template. (See recent commit.)

Comment (lumberyard-employee-dm) [reply content/docs/user-guide/programming/components/create-component.md:19]: Looks good to me.

**Review by lumberyard-employee-dm — APPROVED** [review]

File: content/docs/user-guide/programming/components/system-components.md
Comment (paucom) [inline content/docs/user-guide/programming/components/system-components.md:22]:
```suggestion
For an example of a Gem that provides system components, the `HttpRequestor` Gem that O3DE ships with has an implementation of a system component referenced throughout this topic. The Gem source is located at `Gems/HttpRequestor/Code` in the O3DE source.
```

File: content/docs/user-guide/programming/components/editor-components.md
Comment (paucom) [inline content/docs/user-guide/programming/components/editor-components.md:7]: In the following intro paragraph, change the first occurrence of "O3DE" to "**Open 3D Engine (O3DE)**".

File: content/docs/user-guide/programming/components/editor-components.md
Comment (paucom) [inline content/docs/user-guide/programming/components/editor-components.md:20]:
```suggestion
## Sample editor component
```

File: content/docs/user-guide/programming/components/editor-components.md
Comment (paucom) [inline content/docs/user-guide/programming/components/editor-components.md:72]: In the H2 (##) section header on line 69, change "Editor Component and Runtime Component Differences" to use sentence casing.

File: content/docs/user-guide/programming/components/editor-components.md
Comment (paucom) [inline content/docs/user-guide/programming/components/editor-components.md:73]:
```suggestion
### Base classes
```

File: content/docs/user-guide/programming/components/editor-components.md
Comment (paucom) [inline content/docs/user-guide/programming/components/editor-components.md:108]:
```suggestion
### The DisplayEntityViewport method
```

File: content/docs/user-guide/programming/components/editor-components.md
Comment (paucom) [inline content/docs/user-guide/programming/components/editor-components.md:124]:
```suggestion
In the H3 (###) header on line 123, make "Method" lowercase for sentence casing.
```

**Review by paucom — APPROVED** [review]
* Confirmed that the [changes that JwGedit@ requested](https://github.com/o3de/o3de.org/pull/1494#pullrequestreview-950342121) are resolved.
* Reviewed the [latest additions from willihay@](https://github.com/o3de/o3de.org/pull/1494/commits/2eeb185fbc1ecd40df3be6e74c07c2fbb0b9b3d5).
* Lightly reviewed all other files changed, hence the additional review comments. There are several style, formatting, and other convention issues that I recommend resolving before merging. Nothing critical.

Approved

**Pull request merged**
